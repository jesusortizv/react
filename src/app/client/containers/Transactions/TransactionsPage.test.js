import React from 'react';
import TransactionPage from './TransactionsPage';
import renderer from 'react-test-renderer';
import transactionsData from '../../../../data/transactions.json';

beforeEach(() => {
  window.fetch = jest.fn().mockImplementation(() => Promise.resolve(transactionsData));
});

test('Renders correctly', () => {
  const tree = renderer.create(
    <TransactionPage />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
