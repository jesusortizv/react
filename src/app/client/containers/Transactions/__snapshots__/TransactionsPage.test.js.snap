// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<section>
  <div
    className="header"
  >
    <div
      className="logo"
    >
      <a
        className=""
        onClick={[Function]}
        style={Object {}}
        title="Home"
      >
        <img
          alt="Logo"
          className="Nopa"
          src={
            Object {
              "process": [Function],
            }
          }
        />
      </a>
    </div>
    <div>
      <div
        className="header__transactions"
      >
        <div
          className=""
        >
          <a
            className="button"
            onClick={[Function]}
            style={Object {}}
          >
            Transactions
          </a>
        </div>
      </div>
      <div
        className="header__signin"
      >
        <div
          className=""
        >
          <a
            className="button"
            onClick={[Function]}
            style={Object {}}
          >
            Log In
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="layout"
  >
    <div
      className="Transactions main-content"
    >
      <h1>
        Which bank does this account belong to?
      </h1>
      <p>
        Track all of your payments by connecting as many bank accounts as you'd like to your Nopa
        <br />
        account and get updates on your balance instantly. Plus it's free
      </p>
      <div
        className="Transactions__last"
      >
        <p>
          Your transactions for the last 30 days
        </p>
        <div
          className="Transactions__list"
        >
          <table
            className="table"
          >
            <tbody />
          </table>
        </div>
        <div
          className=""
        >
          <a
            className="button"
            onClick={[Function]}
            style={Object {}}
          >
            Show more
          </a>
        </div>
      </div>
    </div>
  </div>
  <section
    className="partners"
  >
    <div>
      <span>
        Our partners:
      </span>
    </div>
    <div>
      <div>
        <img
          alt="Airbnb"
          className="Nopa"
          src={
            Object {
              "process": [Function],
            }
          }
        />
      </div>
      <div>
        <img
          alt="Metro"
          className="Nopa"
          src={
            Object {
              "process": [Function],
            }
          }
        />
      </div>
      <div>
        <img
          alt="Pariti"
          className="Nopa"
          src={
            Object {
              "process": [Function],
            }
          }
        />
      </div>
      <div>
        <img
          alt="Unshackled"
          className="Nopa"
          src={
            Object {
              "process": [Function],
            }
          }
        />
      </div>
    </div>
  </section>
  <footer
    className="footer"
  >
    <div>
      <p>
        © Nopa Limited 2017 All rights reserved. ‘Nopa’ and the Nopa logo are trade marks of Nopa Limited. Nopa is a member of CIFAS - the UK's leading anti-fraud association, and we are registered with the Office of the Information Commissioner (No. Z879078).
      </p>
      <p>
        Nopa Limited is incorporated in England & Wales (registration number 05197592), with its registered office at 1st Floor, Cottons Centre, Tooley Street, London, SE1 2QG. Nopa Limited is authorised and regulated by the Financial Conduct Authority, and entered on the Financial Services Register under firm registration number 563134.
      </p>
    </div>
  </footer>
</section>
`;
